### PR TITLE
Frontend Language

### PR DESCRIPTION
**Pull request by Kristoff-starling**
This draft PR shouldn't be merged until
* #13 is resolved.
* PersistentVolume support for envoy is added.

### CODE REVIEW

**Comment by Romero027** [comment]
let me know when you want me to take a look!

**Comment by Kristoff-starling** [comment]
Main changes:
* `end_of_stream` is temporarily removed (seems that the testing elements in the control plane repo work fine in ambient mode even if `end_of_stream` is removed)
* The consolidation code is rolled back to the previous version as it seems that the new version cannot merge elements in the correct order.
* New grammar & appnet updates for all elements in ping-pong-app

**Comment by Kristoff-starling** [comment]
@Romero027 I've tested the new frontend language together with the control plane. Sidecar mode and Ambient mode work well while grpc interceptors seem not working as expected (the application is successfully booted but the firewall cannot correctly block messages), but I think it may not be an issue related to the frontend language.
I've pushed another [PR](https://github.com/appnet-org/appnet/pull/1) in the control plane repo, please feel free to have a check :)

**Comment by Romero027** [comment]
Thanks! I will take a look later today

File: compiler/element/frontend/parser.py [hunk 1]
```diff
@@ -2,14 +2,24 @@
...
+    OPEN_PAREN_types = []
+    CLOSE_PAREN_types = []
+    tab_len = 4
 
 
 class ElementParser:
     def __init__(self):
         cwd = pathlib.Path(__file__).parent
-        grammar = open(os.path.join(cwd, "element.lark"), "r").read()
+        grammar = open(os.path.join(cwd, "python_style_grammar.lark"), "r").read()
```
Comment (Romero027) [inline compiler/element/frontend/parser.py:20]: I think we can just call it `element.lark`?

Comment (Romero027) [reply compiler/element/frontend/parser.py:20]: I think we can call this `element.lark` and just keep a copy of the old one for reference.

File: compiler/element/frontend/parser.py [hunk 2]
```diff
@@ -2,14 +2,24 @@
...
         # grammar = open(os.path.join(cwd, "new.lark"), "r").read()
-        self.lark_parser = Lark(grammar, start="start")
+        self.lark_parser = Lark(grammar, start="appnet", postlex=AppNetIndenter())
```
Comment (Romero027) [inline compiler/element/frontend/parser.py:22]: remove line 21?

File: examples/applications/ping-pong-app/ping-pong-app.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ spec:
       labels:
         app: frontend
     spec:
-      # nodeName: h2.hotel.meshbench-pg0.clemson.cloudlab.us # Change Placement if needed
+      nodeName: h1.adn-frontend.meshbench-pg0.clemson.cloudlab.us # Change Placement if needed
```
Comment (Romero027) [inline examples/applications/ping-pong-app/ping-pong-app.yaml:28]: can you change this to `<node-name>` and comment it out?

Comment (Romero027) [reply examples/applications/ping-pong-app/ping-pong-app.yaml:28]: same goes to linie 98 and 168

**Review by Romero027 — APPROVED** [review]
I few nits and feel free to merge!

**Pull request merged**